tests/test_lab_dev/test_circuit_components.py

```diff
@@ -0,0 +1,524 @@
+# Copyright 2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Tests for circuit components. """
+
+# pylint: disable=fixme, missing-function-docstring
+
+import numpy as np
+import pytest
+
+from mrmustard.physics.triples import attenuator_Abc, displacement_gate_Abc, vacuum_state_Abc
+from mrmustard.physics.representations import Bargmann
+from mrmustard.lab_dev.circuit_components import (
+    add_bra,
+    CircuitComponent,
+    AdjointView,
+    DualView,
+)
+from mrmustard.lab_dev.wires import Wires
+
+
+class TestCircuitComponent:
+    r"""
+    Tests ``CircuitComponent`` objects.
+    """
+
+    @pytest.mark.parametrize("x", [0.1, [0.2, 0.3]])
+    @pytest.mark.parametrize("y", [0.4, [0.5, 0.6]])
+    def test_init(self, x, y):
+        name = "my_component"
```

Comment:
This (and all other cases where I initialize a Dgate from the triple) will all be replaced by, e.g., `Dgate(x, y)` when I add gates and states